perllib/FixMyStreet/Map/WMSBase.pm

```diff
@@ -0,0 +1,274 @@
+# FixMyStreet:Map::WMSBase
+# Makes it easier for cobrands to use their own WMS base map.
+# This cannot be used directly; you must subclass it and implement several
+# methods. See, e.g. FixMyStreet::Map::Northamptonshire.
+
+package FixMyStreet::Map::WMSBase;
+
+use strict;
+use Math::Trig;
+use Utils;
+use JSON::MaybeXS;
+
+sub scales {
+    my $self = shift;
+    my @scales = (
+        # A list of scales corresponding to zoom levels, e.g.
+        # '192000',
+        # '96000',
+        # '48000',
+        # etc...
+    );
+    return @scales;
+}
+
+# The copyright string to display in the corner of the map.
+sub copyright {
+    return '';
+}
+
+# A hash of parameters that control the zoom options for the map
+sub zoom_parameters {
+    my $self = shift;
+    my $params = {
+        zoom_levels    => scalar $self->scales,
+        default_zoom   => 7,
+        min_zoom_level => 1,
+        id_offset      => 0,
+    };
+    return $params;
+}
+
+# A hash of parameters used in calculations for map tiles
+sub tile_parameters {
+    my $params = {
+        urls         => [ '' ], # URL of the map tiles, up to the /{z}/{x}/{y} part
+        layer_names  => [ '' ],
+        wms_version => '1.0.0',
+        layer_style  => '',
+        matrix_set   => '',
```

Comment:
This is not a default, it should be format I guess? And 'suffix' next line is not used anywhere either (see other comment).